OsvvmScriptsCreateYamlReports.tcl

```diff
@@ -48,6 +48,80 @@ namespace eval ::osvvm {
 package require fileutil
 
 
+# -------------------------------------------------
+# SanitizeTextForReport
+#   Removes ASCII control characters (except \t, \n, \r) that can break YAML/HTML.
+#   This does not attempt full UTF-8 validation; Tcl strings are Unicode.
+proc SanitizeTextForReport {Text} {
+  if {$Text eq ""} {
+    return ""
+  }
+  # Strip C0 controls excluding tab/newline/carriage return, plus DEL.
+  #   - \x00-\x08, \x0B-\x0C, \x0E-\x1F, \x7F
+  regsub -all {[\x00-\x08\x0B\x0C\x0E-\x1F\x7F]} $Text { } Clean
+  return $Clean
+}
+
+
+# -------------------------------------------------
+# FormatYamlScalar
+#   Return a YAML scalar string that uses native types when safe:
+#     - empty -> null (empty scalar)
+#     - true/false (case-insensitive, also accepts TRUE/FALSE)
+#     - integer / float
+#   Otherwise returns a double-quoted string with minimal escaping.
+#
+proc FormatYamlScalar {Value} {
+  # Treat unset/missing as empty
+  if {$Value eq ""} {
+    return "null"
+  }
+
+  set Value [SanitizeTextForReport $Value]
+  set Trimmed [string trim $Value]
+  if {$Trimmed eq ""} {
+    # Keep whitespace-only values as strings
+    set Escaped [string map [list "\\" "\\\\" "\"" "\\\"" "\n" "\\n" "\r" "\\r" "\t" "\\t"] $Value]
+    return "\"$Escaped\""
+  }
+
+  # null
+  if {[string equal -nocase $Trimmed "null"]} {
+    return "null"
+  }
+
+  # boolean
+  if {[string equal -nocase $Trimmed "true"] || [string equal -nocase $Trimmed "false"]} {
+    return [string tolower $Trimmed]
+  }
+
+  # integer
+  if {[regexp {^[-+]?\d+$} $Trimmed]} {
+    return $Trimmed
+  }
+
+  # float (simple forms, incl exponent)
+  if {[regexp {^[-+]?(?:\d+\.\d*|\d*\.\d+)(?:[eE][-+]?\d+)?$} $Trimmed] || [regexp {^[-+]?\d+(?:[eE][-+]?\d+)$} $Trimmed]} {
+    return $Trimmed
+  }
+
+  # default: quoted string
+  # Use [list] to avoid Tcl list parsing edge cases
+  set Escaped [string map [list "\\" "\\\\" "\"" "\\\"" "\n" "\\n" "\r" "\\r" "\t" "\\t"] $Value]
+  return "\"$Escaped\""
+}
+
+
+# -------------------------------------------------
+# FormatYamlDoubleQuotedScalar
+#   Always returns a double-quoted YAML string (never native/null).
+proc FormatYamlDoubleQuotedScalar {Value} {
+  set Value [SanitizeTextForReport $Value]
+  set Escaped [string map [list "\\" "\\\\" "\"" "\\\"" "\n" "\\n" "\r" "\\r" "\t" "\\t"] $Value]
```

Comment:
Aside from the new VHDL API, the information in the YAML file will not have control characters.    

What is the capability we are trying to gain by adding handling of control characters?